src/renderer/services/chain/transaction/transfer.ts

```diff
@@ -29,7 +32,12 @@ export const transfer$: SendTxStateHandler = (params) => {
         steps: { current: 2, total: 2 }
       })
       // 2. check tx finality by polling its tx data
-      return txStatusByChain$(txHash, params.asset.chain)
+      const assetAddress = isEthChain(params.asset.chain)
+        ? isEthAsset(params.asset)
+          ? undefined
+          : getTokenAddress(params.asset) || undefined
+        : undefined
```

Comment:
[nitpick] Can we do the following? _Just to avoid all these nested `undefined` ..._
```ts
const erc20Address = isEthChain(params.asset.chain) ? getTokenAddress(params.asset) : undefined
```